stix_shifter_modules/reversinglabs/stix_translation/results_translator.py

```diff
@@ -0,0 +1,352 @@
+          elif unknown > 0:
+            tis_range_min = UNKNOWN_SCORE_MIN
+            tis_range_max = UNKNOWN_SCORE_MAX
+            threat_score = int(unknown/known * 10)
+          else:
+            return {"threat_score": 0.0}
+          range_min = 1
+          range_max = 10
+          if threat_score > 10:
+            threat_score = 10
+          tis_score = tis_range_min + (tis_range_max-tis_range_min) * 1.0 * (threat_score-range_min) / (range_max - range_min)
+          return {"threat_score": round(tis_score, 1)}
+
+        # When Hash is passed
+        elif "malware_presence" in data[ds_key][0] and 'threat_level' in data[ds_key][0]['malware_presence'] and 'trust_factor' in data[ds_key][0]['malware_presence']:
+          threat_level = data[ds_key][0]['malware_presence']['threat_level']
+          trust_factor = data[ds_key][0]['malware_presence']['trust_factor']
+          # Since trust factor = 0 is most trustworthy, we flip the list
+          trust_range = [5, 4, 3, 2, 1, 0]
+          trust = trust_range[trust_factor]
+          threat_score = threat_level * trust
+          range_min = 0
+          range_max = 25
+          if(indicator['indicator_types'][0] == 'benign'):
+            tis_range_min = BENIGN_SCORE_MIN
+            tis_range_max = BENIGN_SCORE_MAX
+            tis_score = tis_range_min + (tis_range_max-tis_range_min) * 1.0 * (threat_score-range_min) / (range_max - range_min)
+          elif(indicator['indicator_types'][0] == 'unknown'):
+            tis_range_min = UNKNOWN_SCORE_MIN
+            tis_range_max = UNKNOWN_SCORE_MAX
+            tis_score = tis_range_min + (tis_range_max-tis_range_min) * 1.0 * (threat_score-range_min) / (range_max - range_min)
+          elif (indicator['indicator_types'][0] == 'anomalous-activity'):
+            tis_range_min = SUSPICIOUS_SCORE_MIN
+            tis_range_max = SUSPICIOUS_SCORE_MAX
+            tis_score = tis_range_min + (tis_range_max-tis_range_min) * 1.0 * (threat_score-range_min) / (range_max - range_min)
+          else:
+            tis_range_min = MALICIOUS_SCORE_MIN
+            tis_range_max = MALICIOUS_SCORE_MAX
+            tis_score = tis_range_min + (tis_range_max-tis_range_min) * 1.0 * (threat_score-range_min) / (range_max - range_min)
+
+          threat_score = {"threat_score": round(tis_score, 1)}
+          return threat_score
+
+        else:
+          if(indicator['indicator_types'][0] == 'benign'):
+            return {"threat_score": round(BENIGN_SCORE_MIN, 1)}
+          elif(indicator['indicator_types'][0] == 'unknown'):
+            return {"threat_score": round(UNKNOWN_SCORE_MIN, 1)}
+          elif(indicator['indicator_types'][0] == 'anomalous-activity'):
+            return {"threat_score": round(SUSPICIOUS_SCORE_MIN, 1)}
+          else:
+            return {"threat_score": round(MALICIOUS_SCORE_MIN, 1)}
+
+
+  def translate_results(self, data_source, data):
+    """
+    Translates JSON data into STIX results based on a mapping file
```

Comment:
Is the to-stix mapping file even used? I notice that it's an empty object in the JSON.

Reply:
Its not used, I can update the description

Reply:
Updated description